### PR TITLE
Add Lua helpers for pushing/getting glm types

### PR DESCRIPTION
**Pull request by WeirdRubberDuck**

### CODE REVIEW

File: tests/test_luaconversions.cpp [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+        const T value = ghoul::lua::value<T>(state);
+        REQUIRE(value == val);
+
+        lua_pop(state, 1);
+    }
+
+    lua_close(state);
+}
+
+TEMPLATE_TEST_CASE("LuaConversion Fuzz >short", "[luaconversion]", short, unsigned short,
```
Comment (alexanderbock) [inline tests/test_luaconversions.cpp:129]: Why split the <short and >short test cases?  What am I now seeing?

File: tests/test_luaconversions.cpp [hunk 2]
```diff
@@ -0,0 +1,605 @@
...
+    using T = TestType;
+
+    std::mt19937 gen(1337);
+    std::uniform_real_distribution<typename T::value_type> dis(
+        std::numeric_limits<typename T::value_type>::min(),
+        std::numeric_limits<typename T::value_type>::max()
+    );
+
+    for (int i = 0; i < NumberFuzzTests; ++i) {
+        const T val = T(dis(gen), dis(gen), dis(gen), dis(gen), dis(gen), dis(gen),
```
Comment (alexanderbock) [inline tests/test_luaconversions.cpp:497]: A lot of dissing going on ;)

**Review by alexanderbock — APPROVED** [review]

File: tests/test_luaconversions.cpp (hunk 1 above)
Comment (WeirdRubberDuck) [reply tests/test_luaconversions.cpp:129]: Good question (I basically mainly did adaptions of the old tests). Looks like the only difference is the type passed to the `uniform_int_distribution`? `std::uniform_int_distribution<short>` vs `std::uniform_int_distribution<T>`. Could try combining them

Comment (WeirdRubberDuck) [reply tests/test_luaconversions.cpp:129]: Ah, ofc. The problem is that `uniform_int_distribution` does not accept chars as template argument

**Pull request merged**